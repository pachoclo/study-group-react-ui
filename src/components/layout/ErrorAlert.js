import React from 'react'
import PropTypes from 'prop-types'

function ErrorAlert ({ error, followUp }) {
  return (
    <div className='alert alert-danger alert-dismissible mb-0' role='alert'>
      <i className='icon-left fas fa-exclamation-triangle mr-3' />
      <strong>{error}</strong> <span className='ml-2'>{followUp}</span>
      <button type='button' className='close' data-dismiss='alert' aria-label='Close'>
        <span aria-hidden='true'>&times;</span>
      </button>
    </div>
  )
}

Error.propTypes = {
  error: PropTypes.string,
  followUp: PropTypes.string
}

export default ErrorAlert
