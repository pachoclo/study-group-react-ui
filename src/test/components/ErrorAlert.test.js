import React from 'react'
import ErrorAlert from '../../components/ErrorAlert'
import { shallow } from 'enzyme'

describe('ErrorAlert Test', () => {
  it('should render correctly', () => {
    const component = shallow(<ErrorAlert error={'elError'} />)
    expect(component).toMatchSnapshot()
  })
})
