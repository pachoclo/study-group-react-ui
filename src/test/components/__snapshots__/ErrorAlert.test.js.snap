// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert Test should render correctly 1`] = `
<div
  className="alert alert-danger alert-dismissible mb-0"
  role="alert"
>
  <i
    className="icon-left fas fa-exclamation-triangle mr-3"
  />
  <strong>
    elError
  </strong>
   
  <span
    className="ml-2"
  />
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
</div>
`;
